prow/cmd/phaino/local.go

```diff
@@ -246,23 +254,66 @@ func convertToLocal(ctx context.Context, log *logrus.Entry, pj prowapi.ProwJob,
 		log.Warn("Ignoring resource requirements")
 	}
 
-	for _, mount := range container.VolumeMounts {
-		vol := volume(*pj.Spec.PodSpec, mount.Name)
-		if vol == nil {
-			return nil, fmt.Errorf("mount %q missing associated volume", mount.Name)
+	if opt.mountJobVolumes {
+		for _, mount := range container.VolumeMounts {
+			vol := volume(*pj.Spec.PodSpec, mount.Name)
+			if vol == nil {
+				return nil, fmt.Errorf("mount %q missing associated volume", mount.Name)
+			}
+			if vol.EmptyDir != nil {
+				localArgs = append(localArgs, "-v", mount.MountPath)
+			} else {
+				local, err := readMount(ctx, mount)
+				if err != nil {
+					return nil, fmt.Errorf("bad mount %q: %v", mount.Name, err)
+				}
+				arg := local + ":" + mount.MountPath
+				if mount.ReadOnly {
+					arg += ":ro"
+				}
+				localArgs = append(localArgs, "-v", arg)
+			}
 		}
-		if vol.EmptyDir != nil {
-			localArgs = append(localArgs, "-v", mount.MountPath)
+	}
+
+	// Setup gcloud credentials
+	if opt.useLocalGcloudCredentials {
+		gcloudKey := os.Getenv("GOOGLE_APPLICATION_CREDENTIALS")
+		// If GOOGLE_APPLICATION_CREDENTIALS is not empty, also mount the key file
+		// to the container
+		if gcloudKey != "" {
+			localArgs = append(localArgs, "-e", "GOOGLE_APPLICATION_CREDENTIALS="+gcloudKey)
+			localArgs = append(localArgs, "-v", gcloudKey + ":" + gcloudKey + ":ro")
 		} else {
-			local, err := readMount(ctx, mount)
-			if err != nil {
-				return nil, fmt.Errorf("bad mount %q: %v", mount.Name, err)
+			// We only want to use the default gcloud credentials if GOOGLE_APPLICATION_CREDENTIALS env var is not explicitly set.
+			defaultGcloudConfigPath := path.Join(os.Getenv("HOME"), ".config/gcloud")
+			if _, err := os.Stat(defaultGcloudConfigPath); !os.IsNotExist(err) {
+				localArgs = append(localArgs, "-v", defaultGcloudConfigPath + ":" + gcloudDefaultConfigMountPath)
+				// Overwrite the gcloud config path, as per https://stackoverflow.com/a/48343135
+				localArgs = append(localArgs, "-e", "CLOUDSDK_CONFIG=" + gcloudDefaultConfigMountPath)
+			}
+		}
```

Comment:
There is a chance both these failed, probably deserve a warning here

Reply:
Added the warning log.